pkg/hubble/container/ring.go

```diff
@@ -369,7 +373,26 @@ func (r *Ring) readFrom(ctx context.Context, read uint64, ch chan<- *v1.Event) {
 			case <-ctx.Done():
 				return
 			}
-
+		default:
+			// The writer overwrote the entry before we had time to read it.
+			// Send a ListEvent to notify the read-miss.
+			now := time.Now().UTC()
+			select {
+			case ch <- &v1.Event{
+				Timestamp: &timestamp.Timestamp{
+					Seconds: int64(now.Unix()),
+					Nanos:   int32(now.Nanosecond()),
+				},
```

Comment:
I believe this breaks the "ring buffer timestamps are supposed to be monotonic" assumption. It will cause the ring reader to stop immediately if `until` is set here:

https://github.com/cilium/cilium/blob/8030a81dbbcc189814163742fdb32af740c2cb80/pkg/hubble/observer/local_observer.go#L385-L393

Reply:
Very nice catch! What would you propose we do instead?

Reply:
I gave this more thoughts. I think that lost events need to be treated as a special case, similarly to node status events, as users will never explicitly request them. I think that we should simply send them over whenever they're encountered and not apply any filters on it. This would be more inline with what should be expected and solve the timestamp issue.
I'll update the PR accordingly.

Reply:
@gandro I updated the PR, PTAL.